tests/reprod/fixtures.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def assert_reproducible(func, equals, num_iter=20):
+    """
+    Assert results from calling ``func`` are reproducible. The ``equals`` function is used to test equality of results.
+
+    Args:
+        func (callable): Function to check for reproducible "result"
+        equals (callable): Function to take two arguments (of "result" type) and check that they are equal
```

Comment:
Is it worth making this a bit more structured/restricted? E.g. require that `func` returns a trained keras model, and then do the `get_weights`/`all`-equal thing in here, and not need `equals` at all?

Reply:
Hmm yeah that's probably better, will update